heaps/min_heap.py

```diff
@@ -59,18 +71,25 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: O(log n)
+            Space complexity: O(1)
         """
-        pass
+        if index == 0:
+            return index
+
+        parent_node = (index - 1) // 2
+
+        if self.store[parent_node].key > self.store[index].key:
+            self.swap(parent_node, index)
+            self.heap_up(parent_node)
 
     def heap_down(self, index):
         """ This helper method takes an index and 
             moves the corresponding element down the heap if it's 
             larger than either of its children and continues until
             the heap property is reestablished.
         """
-        pass
+        raise NotImplementedMethodError("Method hasn't been implemented yet!")
```

Comment:
⚠️   Just noting this method isn't written yet.

The trick is:

* Determine if there are left and right children using the formula.
* Find the minimum child
* if the current node is less than the minimum child swap them and call `heap_down` on the new current node